toto/runlib.py

```diff
@@ -177,8 +177,8 @@ def execute_link(link_cmd_args, record_byproducts):
 
   return {"stdout": stdout_str, "stderr": stderr_str}, return_value
 
-def create_link_metadata(link_name, materials_dict, products_dict,
-    link_cmd_args, byproducts, return_value):
+def create_link_metadata(link_name, materials_dict={}, products_dict={},
+    link_cmd_args="", byproducts={}, return_value=None):
```

Comment:
Yeah, I guess this is a non-issue. I can't think of an example in which create_link_metadata is called twice in the same process.